test/pilosa.go

```diff
@@ -192,22 +151,101 @@ func (m *Main) Client() *http.InternalClient {
 }
 
 // Query executes a query against the program through the HTTP API.
-func (m *Main) Query(index, rawQuery, query string) (string, error) {
+func (m *Command) Query(index, rawQuery, query string) (string, error) {
 	resp := MustDo("POST", m.URL()+fmt.Sprintf("/index/%s/query?", index)+rawQuery, query)
 	if resp.StatusCode != gohttp.StatusOK {
 		return "", fmt.Errorf("invalid status: %d, body=%s", resp.StatusCode, resp.Body)
 	}
 	return resp.Body, nil
 }
 
-func (m *Main) RecalculateCaches() error {
+func (m *Command) RecalculateCaches() error {
 	resp := MustDo("POST", fmt.Sprintf("%s/recalculate-caches", m.URL()), "")
 	if resp.StatusCode != 204 {
 		return fmt.Errorf("invalid status: %d, body=%s", resp.StatusCode, resp.Body)
 	}
 	return nil
 }
 
+// Cluster represents a Pilosa cluster (multiple Command instances)
+type Cluster []*Command
+
+// Start runs a Cluster
+func (c Cluster) Start() error {
+	var gossipSeeds = make([]string, len(c))
+	for i, cc := range c {
+		cc.Config.Gossip.Port = "0"
+		cc.Config.Gossip.Seeds = gossipSeeds[:i]
+		if err := cc.Start(); err != nil {
+			return errors.Wrapf(err, "starting server %d", i)
+		}
+		gossipSeeds[i] = cc.GossipTransport().URI.String()
```

Comment:
we can use GossipAddress() here if we want to keep that... otherwise we just need to change everyone using GossipAddress to use GossipTransport().URI.String()